libzmix/benches/bbs_vs_ps.rs

```diff
@@ -29,16 +37,62 @@ fn sign_messages_benchmark(c: &mut Criterion) {
             b.iter(|| BBSSignature::new(attributes.as_slice(), &sk, &pk))
         });
         let label = format!("ps sign {} atts", atts);
-        let (pk, sk) = ps_keys_generate(atts, label.as_bytes());
+        let params = Params::new(label.as_bytes());
+        let (_, sk) = ps_keys_generate(atts, &params);
         c.bench_function(label.as_str(), |b| {
-            b.iter(|| PSSignature::new(attributes.as_slice(), &sk, &pk))
+            b.iter(|| PSSignature::new(attributes.as_slice(), &sk, &params))
         });
     }
 }
 
-fn bbs_sign_committed_messages_benchmark(c: &mut Criterion) {}
+fn bbs_sign_committed_messages_benchmark(c: &mut Criterion) {
+
+}
+
+fn bbs_prove_benchmark(c: &mut Criterion) {
+    for atts in vec![1, 2, 5, 10, 20, 50, 100, 200] {
+        ////////////////////////// BBS+ Signatures
+        let (pk, sk) = bbs_keys_generate(atts).unwrap();
+        let attributes = SignatureMessageVector::random(atts);
+        let sig = BBSSignature::new(attributes.as_slice(), &sk, &pk).unwrap();
+
+        c.bench_function(format!("bbs+ generate proof {} atts", atts).as_str(), |b| {
+            b.iter(|| {
+                let pok = BBSPoKOfSignature::init(&sig, &pk, attributes.as_slice(), None, HashSet::new()).unwrap();
+                let challenge = FieldElement::from_msg_hash(&pok.to_bytes());
+                pok.gen_proof(&challenge).unwrap()
+            })
+        });
+        let poks = BBSPoKOfSignature::init(&sig, &pk, attributes.as_slice(), None, HashSet::new()).unwrap();
+        let challenge = FieldElement::from_msg_hash(&poks.to_bytes());
+        let proof = poks.gen_proof(&challenge).unwrap();
+        c.bench_function(format!("bbs+ verify proof {} atts", atts).as_str(), |b| {
```

Comment:
This benchmark works on the same proof data in each iteration. Are you sure this will not use cached results? A more accurate approach will to collect proofs from previous benchmark in a vector (if you want to exclude time to `push` to a vector, you can use `iter_custom` in place of `iter` and calculate time manually inside it) like this
```
let mut proofs = vec![];
 c.bench_function(format!("bbs+ generate proof {} atts", atts).as_str(), |b| {
            b.iter(|| {
                ...
                let p = pok.gen_proof(&challenge).unwrap();
                proofs.push(p)
```
Then use `iter_batched` to pass the `proofs` to the benchmark.

Reply:
As a maintainer you can update this directly. The point of the benchmark is to time each component. Partially why I do it this way is to eliminate bias in the timing as well as Rust has some ownership complaints if I try stuff like your suggesting. Try it on your end and see what you can get for it

Reply:
It's not caching. If you add a println! to the signature, its different everytime